api/v1alpha1/serviceintegration_types.go

```diff
@@ -103,8 +120,33 @@ type ServiceIntegration struct {
 	Status ServiceIntegrationStatus `json:"status,omitempty"`
 }
 
-func (svcint ServiceIntegration) AuthSecretRef() *AuthSecretReference {
-	return svcint.Spec.AuthSecretRef
+func (in *ServiceIntegration) AuthSecretRef() *AuthSecretReference {
+	return in.Spec.AuthSecretRef
+}
+
+func (in *ServiceIntegration) GetUserConfig() (any, error) {
+	configs := map[string]any{
+		"clickhouse_kafka":                in.Spec.ClickhouseKafkaUserConfig,
+		"clickhouse_postgresql":           in.Spec.ClickhousePostgreSQLUserConfig,
+		"datadog":                         in.Spec.DatadogUserConfig,
+		"external_aws_cloudwatch_metrics": in.Spec.ExternalAWSCloudwatchMetricsUserConfig,
+		"kafka_connect":                   in.Spec.KafkaConnectUserConfig,
+		"kafka_logs":                      in.Spec.KafkaLogsUserConfig,
+		"kafka_mirrormaker":               in.Spec.KafkaMirrormakerUserConfig,
+		"logs":                            in.Spec.LogsUserConfig,
+		"metrics":                         in.Spec.MetricsUserConfig,
+	}
+
+	thisType := in.Spec.IntegrationType
+
+	// Checks if it is the only configuration set
+	for k, v := range configs {
+		if k != thisType && !reflect.ValueOf(v).IsNil() {
+			return nil, fmt.Errorf("got additional configuration for integration type %q", k)
+		}
+	}
+
+	return configs[thisType], nil
 }
```

Comment:
Unfortunately, I can't say for sure which type requires a configuration. E.g. `clickhouse_postgresql` comes with the only configuration field `Databases`, which can be [empty](https://github.com/aiven/aiven-operator/blob/3ff3d75f3d83eb17bc403c43a065359e81094edc/api/v1alpha1/userconfig/integration/clickhouse_postgresql/clickhouse_postgresql.go#L23-L23). Therefore, it is possible to create `clickhouse_postgresql` service integration without paticular configuration.